goose.go

```diff
@@ -1281,6 +1281,18 @@ func endsWithReturn(s ast.Stmt) bool {
 	}
 }
 
+func endIncludesIf(s ast.Stmt) bool {
+	if s == nil {
+		return false
+	}
+	switch s := s.(type) {
+	case *ast.BlockStmt:
+		return stmtsEndIncludesIf(s.List)
```

Comment:
You can use a "type assertion" rather than a type switch for this pattern:

```go
if s, ok := s.(*ast.BlockStmt); ok {
  return stmtsEndIncludesIf(s.List)
}
return stmtsEndIncludesIf([]ast.Stmt{s})